frontend/src/model/releases/EditableRelease.tsx

```diff
@@ -94,24 +107,63 @@ export default function EditableRelease({ release, isEdit, onIsEditChange }: Edi
 
   const handleSubmit = async () => {
     setIsLoading(true)
-
     const fileIds: string[] = []
+    const newFilesToUpload: File[] = []
     for (const file of files) {
+      isFileInterface(file) ? fileIds.push(file._id) : newFilesToUpload.push(file)
+    }
+
+    setFilesToUploadCount(newFilesToUpload.length)
+
+    for (const file of newFilesToUpload) {
       if (isFileInterface(file)) {
         fileIds.push(file._id)
         continue
       }
 
       const metadata = filesMetadata.find((fileWithMetadata) => fileWithMetadata.fileName === file.name)?.metadata
-      const postFileResponse = await postFile(file, model.id, file.name, file.type, metadata)
 
-      if (!postFileResponse.ok) {
-        setErrorMessage(await getErrorMessage(postFileResponse))
+      const fileResponse = await axios
```

Comment:
If we split this logic out as described in another comment we won't have to duplicate the logic as you've done here.